### DISCUSSION TITLE
Standalone Components

### DISCUSSION THREAD

**Discussion by ElCapitanSponge** [primefaces/community#923] [PrimeNG] [answered]
With the release of Angular 17 and the support of this in PrimeNG v17, is there any future plans for moving the Angular code to leverage standalone components (https://angular.dev/guide/components/importing and https://angular.io/guide/standalone-components)?

If this is something that would be likely, the following could aid in this conversion: https://angular.dev/reference/migrations/standalone

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

We have future plans to migrate components standalone, but before that, we're working on passthrough and unstyled mode.

**Reply by ElCapitanSponge** [reply]
Unstyled mode would be helpful. I am assuming it will be something similar to how tailwind UI and shadcn/UI etc handle just enough styling for the component that you can alter it in a user friendly manner or to extend with your own styles?

**Reply by kyjus25** [reply]
It will be equivalent to how it is being done in [PrimeVue](https://tailwind.primevue.org/)'s unstyled mode

https://tailwind.primevue.org/

**Reply by alexciesielski** [reply]
@cetincakiroglu would you accept PRs for refactoring components to be standalone? 

it should not be a breaking change as we could keep the NgModules while having standalone component/directive functoinality. 

we could mark the NgModules as deprecated and remove them with another major version update.

I would be happy to open a couple of PRs for making some directives standalone as I would need that for a project for mine as well.

**Reply by cetincakiroglu** [reply]
Hi,

We would be very happy to have you contribute in this way. Recently, a contributor added the transform attribute feature to all inputs, and we accepted it. Please remember to tag me when you open an Issue and submit a PR. These kinds of contributions greatly help us improve PrimeNG. 

As a side note, please make one at a time instead of doing all at once. Let's review the first one together.
